src/backend/core/services/search_indexers.py

```diff
@@ -0,0 +1,354 @@
+        """
+
+    # pylint: disable-next=too-many-arguments,too-many-positional-arguments
+    def search(self, text, token, visited=(), page=1, page_size=50):
+        """
+        Search for documents in Find app.
+        Ensure the same default ordering as "Docs" list : -updated_at
+
+        Returns ids of the documents
+
+        Args:
+            text (str): Text search content.
+            token (str): OIDC Authentication token.
+            visited (list, optional):
+                List of ids of active public documents with LinkTrace
+                Defaults to settings.SEARCH_INDEXER_BATCH_SIZE.
+            page (int, optional):
+                The page number to retrieve.
+                Defaults to 1 if not specified.
+            page_size (int, optional):
+                The number of results to return per page.
+                Defaults to 50 if not specified.
+        """
+        response = self.search_query(
+            data={
+                "q": text,
+                "visited": visited,
+                "services": ["drive"],
+                "page_number": page,
+                "page_size": page_size,
+            },
+            token=token,
+        )
+
+        return [d["_id"] for d in response]
+
+    @abstractmethod
+    def search_query(self, data, token) -> dict:
+        """
+        Retrieve items from indexation database.
+
+        Must be implemented by subclasses.
+        """
+
+
+class SearchIndexer(BaseItemIndexer):
+    """
+    File indexer that pushes text content from files to La Suite Find app.
+    """
+
+    def to_text(self, item):
+        """
+        Convert a file content into an indexable text.
+        """
+        mimetype = item.mimetype or ""
+
+        if mimetype.startswith("text/"):
+            return default_storage.open(item.file_key, "rb").read().decode()
+
+        raise SuspiciousFileOperation(f"Unrecognized mimetype {mimetype}")
+
+    def has_text(self, item):
+        """
+        Return True if the file mimetype can be converted into text for indexation
+        """
+        mimetype = item.mimetype or ""
+
+        return (
+            item.upload_state == models.ItemUploadStateChoices.READY
+            and item.type == models.ItemTypeChoices.FILE
+            and mimetype
+            and any(mimetype.startswith(allowed) for allowed in self.allowed_mimetypes)
+        )
+
+    def serialize_item(self, item, accesses):
+        """
+        Convert a Document to the JSON format expected by La Suite Find.
+
+        Args:
+            document (Document): The document instance.
+            accesses (dict): Mapping of document ID to user/team access.
+
+        Returns:
+            dict: A JSON-serializable dictionary.
+        """
+        doc_path = str(item.path)
+        is_deleted = bool(item.ancestors_deleted_at or item.deleted_at)
+        content = self.to_text(item) if not is_deleted and self.has_text(item) else ""
```

Comment:
I don't understand the logic here. If the item is deleted or it is not possible to have a text content, you will serialize it with an empty content ? Why you don't ignore it ?